editor/editor_node.cpp

```diff
@@ -4349,15 +4349,25 @@ Ref<Texture2D> EditorNode::get_object_icon(const Object *p_object, const String
 	return _get_class_or_script_icon(p_object->get_class(), scr, p_fallback);
 }
 
-Ref<Texture2D> EditorNode::get_class_icon(const String &p_class, const String &p_fallback) {
+Ref<Texture2D> EditorNode::get_class_icon(const String &p_class, bool p_default_fallback, const String &p_fallback) {
 	ERR_FAIL_COND_V_MSG(p_class.is_empty(), nullptr, "Class name cannot be empty.");
 
 	Ref<Script> scr;
 	if (ScriptServer::is_global_class(p_class)) {
 		scr = EditorNode::get_editor_data().script_class_load_script(p_class);
 	}
 
-	return _get_class_or_script_icon(p_class, scr, p_fallback, true);
+	String fallback = p_fallback;
+	// Use a white node icon if the class derives from Node, and a white object icon otherwise. Use a gray version if the class is abstract.
+	if (p_default_fallback && ClassDB::class_exists(p_class)) {
+		if (ClassDB::is_parent_class(p_class, SNAME("Node"))) {
+			fallback = (ClassDB::is_virtual(p_class) || !ClassDB::can_instantiate(p_class)) ? "NodeDisabled" : "Node";
+		} else {
+			fallback = ClassDB::is_virtual(p_class) ? "ObjectDisabled" : "Object";
+		}
+	}
```

Comment:
I would argue that a similar logic should be used in `get_object_icon`. And for that reason I would put it into `_get_class_or_script_icon` in some way.

Perhaps, we can get rid of the boolean flag that you've added and treat empty `p_fallback` as a request to use some default value. And `p_fallback = ""` could be the default value of the argument in both methods.